### ISSUE TITLE
Bookmark events without logging in.

### ISSUE THREAD

**Issue by HimanshiJain** [systers/volunteers-android#67]
Description:
Bookmark events without logging in.

Acceptance Criteria:
1. User should be able to bookmark(star) events within the app, without registering/ logging in into the application.
2. Tests for the same.

**Comment by vaibhavsingh97** [comment]
@HimanshiJain  can i work on this issue?

**Comment by pandilladivya** [comment]
@vaibhavsingh97 
You can work on Bookmarking without login, I have worked on Bookmarking events after the user has logged in as part of GSoC. I will be pushing the code tonight.
So you can confirm with @HimanshiJain, claim this issue and work on it after I push those changes ☺️

**Comment by HimanshiJain** [comment]
Sure @vaibhavsingh97 you can work on this issue after @pandilladivya has pushed with login feature. And till then we can discuss what approach do you plan to use for the same. :)

**Comment by vaibhavsingh97** [comment]
@pandilladivya I have to wait till #80 will merge right?
@HimanshiJain we can have flag which shows user has registered or not. btw i don't understand the need of tracking the registration as bookmarking is a common event for both categories of users

**Comment by pandilladivya** [comment]
@vaibhavsingh97 Yes, I think you need to wait.
How are you going to store the bookmarked events even if the user is not logged in? Can you state your logic in detail.

**Comment by HimanshiJain** [comment]
Yes @vaibhavsingh97 , so bookmarking is a common thing, that user should be able to do without registering or logging in. Tracking registration would be not useful as far as I think.
And we can discuss approaches to things that @pandilladivya has mentioned.

**Comment by vaibhavsingh97** [comment]
@HimanshiJain @pandilladivya I will share detailed description on how i am going to proceed in a day or two.

**Comment by vaibhavsingh97** [comment]
@HimanshiJain @pandilladivya to bookmark events we will use local db to store the events. As the bookmarking is the common event so we don't have to track that user is logged in or not. We will make separate table in database which will store favorite events
- If user uninstalls the application 
A: Database will be deleted so user will not able to see the bookmark events if user reinstall the application

**Comment by sandarumk** [comment]
@pandilladivya Any update about this?

**Comment by pandilladivya** [comment]
@vaibhavsingh97 There is no local db as of now in the application. We are using firebase for everything. 
What is the local db you want to use? SQLite ?

**Comment by HimanshiJain** [comment]
@vaibhavsingh97 you can start working on this now that #82 has been merged.
